qmake/builddir.pri

```diff
@@ -0,0 +1,52 @@
+# Copyright 2005-2018 The Mumble Developers. All rights reserved.
+# Use of this source code is governed by a BSD-style license
+# that can be found in the LICENSE file at the root of the
+# Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+# Globally use "/debug" or "/release"
+# as build directory, if it's not
+# a shadow build, otherwise use
+# the shadowed directory.
+#
+# https://wiki.qt.io/QMake-top-level-srcdir-and-builddir
+
+# Since this file is in the "qmake" directory,
+# "$$PWD/.." is the top of the source tree.
+# absolute_path() is used to clean the path.
+TOP_SRCDIR = $$absolute_path($$PWD/..)
```

Comment:
Should I replace `$$absolute_path` with `$$clean_path`?

I used the first because I thought it was available on Qt 4, but it turned out it isn't.

Reply:
Which ones of those are available in Qt 4?

Reply:
Neither.

Reply:
Then feel free to use clean_path.

I suppose builddir.pri will cause some new warnings on Qt 4 when built with qmake -Wall?

Reply:
From Travis CI:
```
/home/travis/build/mumble-voip/mumble/qmake/builddir.pri:16: Unknown replace function: absolute_path
/home/travis/build/mumble-voip/mumble/qmake/builddir.pri:23: Unknown replace function: shadowed
```